src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/AvoidSingleUseOfLocalJsonSerializerOptionsTests.cs

```diff
@@ -0,0 +1,574 @@
+        public Task CS_UseNewLocalOptionsAsArgument_MultiAssignment_NotSingleUse_NoWarn(string expression1, string expression2)
+            => VerifyCS.VerifyAnalyzerAsync($@"
+using System.Text.Json;
+
+class Program
+{{
+    static string Serialize<T>(T value)
+    {{
+        JsonSerializerOptions opt1, opt2, opt3;
+        opt1 = opt2 = opt3 = new JsonSerializerOptions();
+
+        string json1 = JsonSerializer.Serialize(value, {expression1});
+        string json2 = JsonSerializer.Serialize(value, {expression2});
+        
+        return json1 + json2;
+    }}
+}}
+");
+
+        [Theory]
+        [InlineData("opt1")]
+        [InlineData("opt2")]
+        [InlineData("opt3")]
+        public Task CS_UseNewLocalOptionsAsArgument_MultiAssignment_EscapeCurrentScope_FieldAssignment_NoWarn(string expression)
+            => VerifyCS.VerifyAnalyzerAsync($@"
+using System.Text.Json;
+
+class Program
+{{
+    static JsonSerializerOptions s_options;    
+
+    static string Serialize<T>(T value)
+    {{
+        JsonSerializerOptions opt1, opt2, opt3;
+        opt1 = opt2 = opt3 = new JsonSerializerOptions();
+
+        s_options = {expression};
+
+        return JsonSerializer.Serialize(value, opt1);   
+    }}
+}}
+");
+
+        [Theory]
+        [InlineData("opt1 = opt2 = s_options")]
+        [InlineData("opt1 = s_options = opt2")]
+        [InlineData("s_options = opt1 = opt2")]
+        public Task CS_UseNewLocalOptionsAsArgument_MultiAssignment_EscapeCurrentScope_FieldInMultiAssignment_NoWarn(string expression)
+            => VerifyCS.VerifyAnalyzerAsync($@"
+using System.Text.Json;
+
+class Program
+{{
+    static JsonSerializerOptions s_options;    
+
+    static string Serialize<T>(T value)
+    {{
+        JsonSerializerOptions opt1, opt2;
+        {expression} = new JsonSerializerOptions();
+
+        return JsonSerializer.Serialize(value, opt1);   
+    }}
+}}
+");
+
+        [Theory]
+        [InlineData("s_options = opt1 = opt2")]
+        [InlineData("opt1 = s_options = opt2")]
+        public Task CSharpUseNewOptionsAsLocalThenAsArgument_AssignmentOnNextStatement_Multiple_WithEscapeScopeOnAssignment_NoWarn(string expression)
+            => VerifyCS.VerifyAnalyzerAsync($@"
+using System.Text.Json;
+
+class Program
+{{
+    static JsonSerializerOptions s_options;
+
+    static string Serialize<T>(T value)
+    {{
+        JsonSerializerOptions opt1, opt2;
+        opt1 = opt2 = new JsonSerializerOptions();
+
+        {expression};
+
+        return JsonSerializer.Serialize(value, opt1);
+    }}
+}}
+");
+
+        [Fact]
+        public Task CS_UseNewLocalOptionsAsArgument_EscapeCurrentScope_ClosureDelegate_NoWarn()
+            => VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+using System.Text.Json;
+
+class Program
+{
+    static Action Serialize<T>(T value)
+    {
+        JsonSerializerOptions opt = new JsonSerializerOptions();
+        Action lambda = () =>
+        {
+            JsonSerializer.Serialize(value, opt);
```

Comment:
Wonder why it should not warn here ...

Reply:
Because `opt` is initialized outside of `lambda` and once used in lambda it can escape current scope.

This is mentioned in the PR description:
> If any of the locals _**potentially**_ escapes current block scope, we will bail.

Reply:
Well, this just looks like false negative to me, I guess I am not understanding the point of this analyzer.

Reply:
We don't want the analyzer to catch all possible single uses, just the 80% of cases that are very common.